apps/language_server/lib/language_server/providers/formatting.ex

```diff
@@ -26,25 +27,30 @@ defmodule ElixirLS.LanguageServer.Providers.Formatting do
         :error ->
           {:error, :internal_error, "Unable to fetch formatter options"}
       end
-    else
-      msg =
-        "Cannot format file from current directory " <>
-          "(Currently in #{Path.relative_to(File.cwd!(), project_dir)})"
-
-      {:error, :internal_error, msg}
-    end
+    end)
   rescue
     _e in [TokenMissingError, SyntaxError] ->
       {:error, :internal_error, "Unable to format due to syntax error"}
   end
 
-  # If in an umbrella project, the cwd might be set to a sub-app if it's being compiled. This is
-  # fine if the file we're trying to format is in that app. Otherwise, we return an error.
-  defp can_format?(file_uri, project_dir) do
-    file_path = file_uri |> SourceFile.path_from_uri() |> Path.absname()
+  def try_format(func) do
+    if Mix.Project.umbrella?() do
```

Comment:
Why not lock non-umbrella also?

Reply:
Good question. From what I can understand the bug is only happening with umbrella apps. It sounds like the current working directory changes from app to app as compilation occurs. If the cwd is in a different application than the file being formatted there's an error trying to get the formatter.exs.

So I wanted to avoid acquiring a lock if we don't have to, my thinking is the more we do that in the code base the higher the odds of contention when it's time for a new build. 

What do you think?

Reply:
The error logs indicate that it changes also when compiling deps in regular apps (see https://github.com/elixir-lsp/elixir-ls/issues/252#issue-620473784, https://github.com/elixir-lsp/elixir-ls/issues/381#issuecomment-708322038)

Reply:
Ah I see! I missed the deps part, thanks for pointing it out. Sounds like we'll need to do this for all projects then as you suggest.